docker/Dockerfile

```diff
@@ -1,7 +1,21 @@
+# Stage -1: Update apt-get
+FROM ubuntu:jammy AS base-ubuntu
+
+# Mount apt's cache folders to cache install but maintain
+# reproducibility on local machine. More details at:
+# https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/reference.md#run---mounttypecache
+
+# Set sharing=locked to avoid race condition with between stages.
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update && apt-get upgrade -y
```

Comment:
Do we need this? Is this for upgrading the packages inside the base image?

This partially ruins reproducibility but nyaeh, whatever, it's Docker :cat2:

Reply:
Compared to the previous version of this Dockerfile, this doesn't affect reproducibility at all because this just put apt's cache files into cache. How it is used is up to apt, and I guess it wouldn't behave differently. That is, if I understand apt's behavior correctly :shrug: